server/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -661,4 +717,111 @@ private AuthDataSet executeListUserPrivileges(IAuthorizer authorizer, String use
     }
     return dataSet;
   }
+
+  /**
+   * Set storage group when creating schema automatically is enable
+   */
+  private void setStorageGroupToMTreeByAutoLevel(String deviceId) throws ProcessorException, PathErrorException {
+    IoTDBConfig conf = IoTDBDescriptor.getInstance().getConfig();
+    MetadataPlan metadataPlan = new MetadataPlan(
+        MetadataOperator.NamespaceType.SET_STORAGE_GROUP,
+        new Path(mManager.getStorageGroupNameByAutoLevel(deviceId, conf.getAutoStorageGroupLevel())));
+    operateMetadata(metadataPlan);
+  }
+
+  private void addPathListToMTree(String deviceId, String[] measurementList, Object[] values)
+          throws ProcessorException {
+    for (int i = 0; i < measurementList.length; i++) {
+      addPathToMTree(deviceId, measurementList[i], values[i]);
+    }
+  }
+
+  private void addPathToMTree(String deviceId, String measurementId, Object value)
+          throws ProcessorException {
+    TSDataType predictedDataType = getPredictedDataType(value);
+    TSEncoding defaultEncoding = getDefaultEncoding(predictedDataType);
+    MetadataPlan metadataPlan = new MetadataPlan(
+        MetadataOperator.NamespaceType.ADD_PATH,
+        new Path(deviceId + "." + measurementId),
+        predictedDataType, defaultEncoding,
+        CompressionType.valueOf(TSFileDescriptor.getInstance().getConfig().getCompressor()));
+    operateMetadata(metadataPlan);
+  }
+
+  /**
+   * Get predicted DataType of the given value
+   */
+  private TSDataType getPredictedDataType(Object value) {
+    if (value instanceof Boolean) {
```

Comment:
Under what circumstances will the value object be Boolean or Number? Is there any test code to cover this line?

Reply:
I think this is possible when using a batched insert.